TasksApp/TasksApp/ViewControllers/ContainerViewController.swift

```diff
@@ -0,0 +1,151 @@
+    
+    private func configureChildsControllers() {
+        addChild(menuViewController)
+        view.addSubview(menuViewController.view)
+        menuViewController.didMove(toParent: self)
+        menuViewController.delegate = self
+        
+        let navigationVC = UINavigationController(rootViewController: homeViewController)
+        addChild(navigationVC)
+        view.addSubview(navigationVC.view)
+        navigationVC.didMove(toParent: self)
+        self.navigationVC = navigationVC
+        homeViewController.delegate = self
+    }
+    
+    private func configureGestures() {
+        let swipeLeftGesture = UISwipeGestureRecognizer(target: self, action: #selector(toggleMenu))
+        swipeLeftGesture.direction = .left
+        swipeLeftGesture.cancelsTouchesInView = false
+        view.addGestureRecognizer(swipeLeftGesture)
+        
+        let swipeRightGesture = UISwipeGestureRecognizer(target: self, action: #selector(toggleMenu))
+        swipeRightGesture.direction = .right
+        swipeRightGesture.cancelsTouchesInView = false
+        view.addGestureRecognizer(swipeRightGesture)
+    }
+    
+    private func open(_ task: UIViewController) {
+        homeViewController.addChild(task)
+        homeViewController.view.addSubview(task.view)
+        task.view.frame = homeViewController.view.frame
+        task.didMove(toParent: homeViewController)
+        homeViewController.title = task.title
+        tasksControllers.append(task)
+    }
+    
+    private func resetToHome() {
+        tasksControllers.forEach { taskVC in
+            taskVC.view.removeFromSuperview()
+            taskVC.didMove(toParent: nil)
+        }
+        homeViewController.title = Constants.homeViewControllerTitle
+    }
+    
+    @objc private func toggleMenu() {
+        switch menuState {
+        case .closed:
+            UIView.animate(withDuration: 0.5,
+                           delay: 0, usingSpringWithDamping: 0.8,
+                           initialSpringVelocity: 0,
+                           options: .curveEaseInOut) {
+                
+                self.navigationVC?.view.frame.origin.x = self.homeViewController.view.frame.size.width - 100
+                
+            } completion: { [weak self] _ in
+                self?.menuState = .opened
+            }
+            
+        case .opened:
+            UIView.animate(withDuration: 0.5,
+                           delay: 0,
+                           usingSpringWithDamping: 0.8,
+                           initialSpringVelocity: 0,
+                           options: .curveEaseInOut) {
+                
+                self.navigationVC?.view.frame.origin.x = 0
+                
+            } completion: { [weak self] _ in
+                self?.menuState = .closed
+            }
+        }
+    }
+}
+
+// MARK: - HomeViewControllerDelegate
+
+extension ContainerViewController: HomeViewControllerDelegate {
+    
+    func didTappedButtonMenu() {
```

Comment:
didTapButtonMenu

Reply:
don't quite see why would we wrap toggleMenu in this didTapButtonMenu

Reply:
Processing of user events is handled by ContainerViewController (it contains children HomeViewController and MenuViewController)
When you click on the menu button in the HomeViewController navigation bar, we use the didTapButtonMenu() delegate method to inform the ContainerViewController that the user has pressed the menu button, after which, using the toggleMenu() method, the ContainerViewController animatedly redraws the HomeViewController frame and “pushes it to the side” and shows the side menu - MenuViewController